File: server/app/schemas/conversation_read_status.py
"""
Conversation read-status setting schemas.
"""
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict


class ConversationReadStatusPayload(BaseModel):
    model_config = ConfigDict(extra="forbid")

    agent_workspace_enabled: bool = True
    web_sdk_enabled: bool = True


class ConversationReadStatusResponse(ConversationReadStatusPayload):
    id: int | None = None
    tenant_id: int | None = None
    configured: bool
    updated_by_id: int | None = None
    updated_by_name: str | None = None
    updated_at: datetime | None = None


class ConversationReadStatusTargetResponse(BaseModel):
    target: Literal["agent_workspace", "web_sdk"]
    configured: bool
    enabled: bool
    updated_at: datetime | None = None


class ConversationReadStatusPublicResponse(BaseModel):
    web_sdk_enabled: bool = True

File: web/models/conversation-read-status.ts
export type ConversationReadStatusPayload = {
  agent_workspace_enabled: boolean
  web_sdk_enabled: boolean
}

export type ConversationReadStatusConfig = ConversationReadStatusPayload & {
  id: number | null
  tenant_id: number | null
  configured: boolean
  updated_by_id: number | null
  updated_by_name: string | null
  updated_at: string | null
}

export type ConversationReadStatusTargetConfig = {
  target: 'agent_workspace' | 'web_sdk'
  configured: boolean
  enabled: boolean
  updated_at: string | null
}

export type ConversationReadStatusPublicConfig = {
  web_sdk_enabled: boolean
}
